.pre-commit-config.yaml

```diff
@@ -10,6 +10,8 @@ repos:
         args: [--fix, --show-fixes]
         types_or: [python, pyi, jupyter]
 
+      - id: ruff-format
```

Comment:
```suggestion
      - id: ruff-format
        types_or: [python, pyi, jupyter]
```

Reply:
If i do something like this 

```
 - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: "v0.1.3"
    hooks:
      - id: ruff
        args: [--fix, --show-fixes]

      - id: ruff-format
        types_or: [python, pyi, jupyter]
```

then run the pre-commit locally then it is changing jupyter notebooks and removing `;`.
```
 - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: "v0.1.3"
    hooks:
      - id: ruff
        args: [--fix, --show-fixes]
        types_or: [python, pyi, jupyter]

      - id: ruff-format
 ```
so better we keep this as it is not changing .ipynb files. I have tested it locally twice.

Reply:
```
  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: "v0.1.3"
    hooks:
      - id: ruff
        args: [--fix, --show-fixes]
   

      - id: ruff-format
        types_or: [python, pyi, jupyter]

```
It is removing `;` from .ipynb

Reply:
I think a lot of the semi-colons just stop output like:
```
<pybamm.solvers.solution.Solution at 0x13cc02f10>
```
This does not seem like a huge problem.

Another way to resolve that printing is to save output objects to a discarded variable:
```python
_ = sim.solve()
```
instead of
```python
sim.solve();
```

Reply:
yes, `_ = sim.solve()` this is in the jupyter notebook

But for plotting graphs without  line like this `[<matplotlib.lines.Line2D at 0x7f4123ea7ac0>]` we need semi-colons.

---
**One Solution is to put `# fmt: off` and `# fmt: on` where  semi-colons appear**
```
# fmt: off
plt.plot(sol["Time [s]"].data, sol["Voltage [V]"].data);
# fmt: on
``` 
Should I use these comments or try to find something else?

Reply:
Personally I would prefer to avoid adding tool specific comments if there are other ways.

In this case:
```python
_ = plt.plot(sol["Time [s]"].data, sol["Voltage [V]"].data)
```
works since plot() also has a return

Reply:
Hello @Saransh-cpp 
what changes you prefer ?

Reply:
The semicolon thing was fixed in ruff upstream. You might have to reset your commits and introduce ruff format in an entirely new commit, given that we want all the pre-commit related changes bundled in a single commit.